src/sort.rs

```diff
@@ -119,8 +133,14 @@ pub fn sort_toml(
         if toml.as_table().contains_key(heading) {
             if let Item::Table(table) = &mut toml[heading] {
                 if table.contains_key(key) {
-                    if let Item::Value(Value::Array(arr)) = &mut table[key] {
-                        arr.sort();
+                    match &mut table[key] {
+                        Item::Value(Value::Array(arr)) => {
+                            arr.sort();
+                        }
+                        Item::Table(table) => {
+                            sort_table(table, group);
```

Comment:
I verified that if. leave out this new line of code (`sort_table`), the new test fails - so the test is covering this correctly.